app/controllers/api/v1/activities_controller.rb

```diff
@@ -0,0 +1,23 @@
+class Api::V1::ActivitiesController < ApplicationController
+  protect_from_forgery unless: -> { request.format.json? }
+  before_action :authenticate_user!
+
+  def index
+    workout = Workout.find(params[:workout_id])
+    render json: Activity.where(workout: workout)
+  end
+
+  def update
+    params.permit!
+    params[:activityInfo].each do |key, value|
+      update_activity(key, value)
+    end
+    render json: { data: "saved" }
+  end
```

Comment:
hmmm params.permit! is a pretty bold whitelist. If there is a way we can use normal strong_params, this is much preferred